clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+	numberOfServers   int
+	numberOfClients   int
+	podRatioType      string
+	testDuration      int
+	protocol          string
+	templateMapping   map[string]interface{}
+	// workerPodList stores list of podData for every worker node.
+	podInfo workerPodInfo
+	// metricVal stores MetricResponse received from worker node.
+	metricVal  [][]float64
+	metricLock sync.Mutex
+	stopCh     chan struct{}
+	// startTimeStampForTestExecution is a futureTimeStamp sent to all pods to start
+	// sending the traffic at the same time.
+	startTimeStampForTestExecution int64
+	resultSummary                  testResultSummary
+}
+
+func (npm *networkPerformanceMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+	switch action {
+	case "start":
+		if err = npm.validate(config); err != nil {
+			return nil, err
+		}
+		return nil, npm.start(config)
+	case "gather":
+		summary, err := npm.gather()
+		if err != nil {
+			return nil, err
+		}
+		return []measurement.Summary{summary}, err
+	default:
+		return nil, fmt.Errorf("unknown action")
+	}
+}
+
+func (npm *networkPerformanceMeasurement) start(config *measurement.Config) error {
+	npm.initialize(config)
+	if err := npm.prepareCluster(); err != nil {
+		return err
+	}
+	if err := npm.createAndWaitForWorkerPods(); err != nil {
+		return err
+	}
+	if err := npm.storeWorkerPods(); err != nil {
+		return err
+	}
+	if err := npm.initializeInformer(); err != nil {
+		return err
+	}
+	switch npm.podRatioType {
+	case oneToOne, manyToMany:
+		npm.execNToMTest()
+	default:
+		return fmt.Errorf("invalid Pod Ratio")
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) initialize(config *measurement.Config) {
+	npm.k8sClient = config.ClusterFramework.GetClientSets().GetClient()
+	npm.framework = config.ClusterFramework
+	npm.dynamicClient = config.ClusterFramework.GetDynamicClients().GetClient()
+	npm.podInfo.workerPodMap = make(map[string]podList)
+}
+
+func (npm *networkPerformanceMeasurement) initializeInformer() error {
+	gvr, _ := meta.UnsafeGuessKindToResource(crGvk)
+	npm.resourceInterface = npm.dynamicClient.Resource(gvr).Namespace(netperfNamespace)
+	informer, err := getInformer(netperfNamespace, npm.dynamicClient, gvr)
+	if err != nil {
+		return fmt.Errorf("error getting informer:%s", err)
+	}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			npm.handleUpdateNetworkTestEvents(newObj)
+		},
+	})
+	npm.stopCh = make(chan struct{})
+	go informer.Run(npm.stopCh)
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) prepareCluster() error {
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+	if err := npm.framework.ApplyTemplatedManifests(clusterRoleBindingFilePath, npm.templateMapping); err != nil {
```

Comment:
templateMapping is nil here anyway - just pass nil explicitly
same below

Reply:
modified